usr/src/uts/common/fs/zfs/dsl_scan.c

```diff
@@ -85,13 +161,141 @@ extern int zfs_txg_timeout;
  */
 boolean_t zfs_free_bpobj_enabled = B_TRUE;
 
+static kmem_cache_t *scan_seg_cache;
+
 /* the order has to match pool_scan_type */
 static scan_cb_t *scan_funcs[POOL_SCAN_FUNCS] = {
 	NULL,
 	dsl_scan_scrub_cb,	/* POOL_SCAN_SCRUB */
 	dsl_scan_scrub_cb,	/* POOL_SCAN_RESILVER */
 };
 
+typedef struct {
+	range_seg_t	ss_rs;
+	avl_node_t	ss_size_node;
+	uint64_t	ss_fill;
+} scan_seg_t;
+
+typedef struct scan_io {
+	dva_t			sio_dva;
+	uint64_t		sio_prop;
```

Comment:
this is a copy of `blk_prop`, right?  Maybe we should name it `sio_blk_prop` to make that extra clear (as opposed to some other properties).